src/ProjectOrigin.VerifiableEventStore/Services/Repository/InMemory/InMemoryRepository.cs

```diff
@@ -47,13 +49,15 @@ public class InMemoryRepository : ITransactionRepository
             };
 
             var blockHash = BlockHash.FromHeader(blockHeader);
-            _blocks[blockHash] = new BlockRecord(blockHeader, null, fromTransaction, fromTransaction + numberOfTransactions - 1);
+            var newBlock = new BlockRecord(blockHeader, null, fromTransaction, fromTransaction + numberOfTransactions - 1);
+            _blocks[blockHash] = newBlock;
+            _blockList.Add(newBlock);
```

Comment:
The use of both a dictionary (_blocks) and a list (_blockList) to store BlockRecord, is interesting. Is the order of the blocks important to maintain? If not, consider how performance is affected, with a big collection. Have you considered, synchronization concerns, in multi-threaded environments?